src/hooks/useQuota/useQuota.tsx

```diff
@@ -0,0 +1,114 @@
+import { useState } from "react";
+import { Sentry } from "../../utils/errorTracking";
+import { getQuota, QuotaError, NotEligibleError } from "../../services/quota";
+import { useProductContext } from "../../context/products";
+
+import { transform } from "lodash";
+
+import { Quota, Policy } from "../../types";
+
+export type QuotaHook = {
+  quotaResponse: Quota | null;
+  allQuotaResponse: Quota | null;
+  fetchQuota: (setter: (cartState: CartState) => void) => Promise<void>;
+  quotaError?: Error;
+};
+
+type CartState =
```

Comment:
What's the reason for letting `useQuota` know about the `CartState`? How might we structure this hook so that it is agnostic on where it's called, and stays true to its single responsibility?

Reply:
This no longer exists in the code, let me know if this is still an issue.